tests/components/fyta/test_init.py

```diff
@@ -1,23 +1,102 @@
 """Test the initialization."""
 
-from unittest.mock import AsyncMock
+from datetime import UTC, datetime
+from unittest.mock import AsyncMock, patch
 
-from homeassistant.components.fyta.const import CONF_EXPIRATION, DOMAIN
+from fyta_cli.fyta_exceptions import (
+    FytaAuthentificationError,
+    FytaConnectionError,
+    FytaPasswordError,
+)
+import pytest
+
+from homeassistant.components.fyta.const import CONF_EXPIRATION, DOMAIN as FYTA_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntryState
 from homeassistant.const import CONF_ACCESS_TOKEN, CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
 
-from .test_config_flow import ACCESS_TOKEN, PASSWORD, USERNAME
+from . import setup_platform
+from .const import ACCESS_TOKEN, EXPIRATION, EXPIRATION_OLD, PASSWORD, USERNAME
 
 from tests.common import MockConfigEntry
 
 
+async def test_load_unload(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_fyta_connector: AsyncMock,
+) -> None:
+    """Test load and unload."""
+
+    await setup_platform(hass, mock_config_entry, [SENSOR_DOMAIN])
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.NOT_LOADED
+
+
+@pytest.mark.parametrize(
+    "exception",
+    [
+        FytaAuthentificationError,
+        FytaPasswordError,
+    ],
+)
+async def test_invalid_credentials(
+    hass: HomeAssistant,
+    exception: Exception,
+    mock_config_entry: MockConfigEntry,
+    mock_fyta_connector: AsyncMock,
+) -> None:
+    """Test FYTA credentials changing."""
+
+    mock_fyta_connector.expiration = datetime.fromisoformat(EXPIRATION_OLD).replace(
+        tzinfo=UTC
+    )
+    mock_fyta_connector.login.side_effect = exception
+
+    with patch(
+        "homeassistant.components.fyta.config_flow.FytaConfigFlow.async_step_reauth",
+        return_value={
+            "type": FlowResultType.ABORT,
+            "flow_id": "mock_flow",
+            "step_id": "reauth_confirm",
+            "reason": "reauth_successful",
+        },
+    ) as mock_async_step_reauth:
+        await setup_platform(hass, mock_config_entry, [SENSOR_DOMAIN])
+        await hass.async_block_till_done()
```

Comment:
We should not patch the config flow, we should just assert this sets the state of the entry to SETUP_ERROR and starts a new flow

Reply:
Got this from https://github.com/home-assistant/core/blob/3efdeaaa778351018b6c55f467774e576165f209/tests/components/abode/test_init.py#L73

Reply:
I see `assert mock_config_entry.state is ConfigEntryState.SETUP_ERROR`.
How do I test if a new flow is started? Or is the first sufficient?